pkg/operator/ceph/object/rgw.go

```diff
@@ -317,3 +318,27 @@ func buildDNSEndpoint(domainName string, port int32, secure bool) string {
 	}
 	return fmt.Sprintf("%s://%s:%d", httpPrefix, domainName, port)
 }
+
+// GetTLSCACert fetch cacert for internal RGW requests
+func GetTLSCACert(objContext *Context, objectStoreSpec *cephv1.ObjectStoreSpec) ([]byte, error) {
+	ctx := context.TODO()
+	var (
+		tlsCert []byte
+		err     error
+	)
+
+	if objectStoreSpec.Gateway.SSLCertificateRef != "" {
+		tlsSecretCert, err := objContext.Context.Clientset.CoreV1().Secrets(objContext.clusterInfo.Namespace).Get(ctx, objectStoreSpec.Gateway.SSLCertificateRef, metav1.GetOptions{})
+		if err != nil {
+			return nil, err
```

Comment:
What additional context can you add to the error to add clarity? E.g., `failed to get secret "<objectStoreSpec.Gateway.SSLCertificateRef>" defined in object store "<objectStoreSpec.name>" gateway.sslCertificateRef`